### DISCUSSION TITLE
[FEATURE REQUEST] Incremental CFG value

### DISCUSSION THREAD

**Discussion by Michoko92** [hlky/stable-diffusion-webui#259] [Feature suggestions]
Hi,

Something that I do a lot, when I finally found an interesting seed, is to make CFG variations to see what looks best. I go incrementally, like 7, 8, 9, 10... I think it would be awesome if this could be done programmatically.

Basically, it would mean adding an "Incremental CFG value" field that would be added to each image generation of the same batch. So if you set this to 0.5, it would generate images with the same seed, but adding progressively 0.5 to the CFG Scale value for each image. 

Example: With an iteration value of 8, a starting CFG at 7 and an incremental CFG value of 1, we would then have variations with CFG = 7, 8, 9... up to 15. That would such a time saver!

Thank you for your awesome work!

**Comment by kalkal11** [comment]
Same with steps too.

Imagine automatic grids like with the matrix option but for scale vs steps (within a set range/increments of course so it doesn't go out of hand)

**Reply by Hamalittle** [reply]
I have one prompt where the subject changes genders in as little as a simple sampling step. And weirdly changing sampling steps while changing CFG can result in very very similar results while changing just the steps changes quite a bit.

**Comment by e850205** [comment]
Both of these would be great

**Comment by Michoko92** [comment]
I think that having two parameters then, one for incremental CFG, and one for incremental steps would be clearer than messing with matrices, which can be sometimes difficult to figure out for non technical people. It's easier to implement too.

**Reply by kalkal11** [reply]
I'm not sure I can agree with that, combining changes of steps and cfg is an advanced usage in its own right and anything that can provide a visual indication of the differences both of them achieve should simplify leaning imo

I can also see the merit to having both of an option, but also, adding options increases complexity to and leads to ui clutter.

**Comment by listelin** [comment]
It has already been implemented here.
https://github.com/AUTOMATIC1111/stable-diffusion-webui

It also has other very useful features.
